website_sale_product_detail_attribute_image/controllers/main.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2019 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+from odoo import http
+
+
+class WebsiteSaleProductDetailAttributeImage(WebsiteSale):
+
+    @http.route()
+    def product(self, product, category='', search='', **kwargs):
+        res = super().product(
+            product, category=category, search=search, **kwargs)
+        image_attributes = product.attribute_line_ids.filtered(
+            lambda x: x.attribute_id.website_product_detail_display)
```

Comment:
Isn't better to populate images directly:
```python
attribute_images = product.attribute_line_ids.filtered(
    lambda x: x.attribute_id.website_product_detail_display
).mapped('website_product_detail_image')
```

Reply:
I self-answer: you are also displaying the attribute name. Not sure about this being coupled between backend and frontend. It's better to have an extra translate field for website display, but fallbacking to this one if not set.